common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -354,6 +378,19 @@ public boolean close() {
         }
 
         @Override
+        public boolean close(T trackedObject) {
+            // We need to actually do the null check of the trackedObject after we close the leak because otherwise
+            // we may get false-positives reported by the ResourceLeakDetector. This can happen as the JIT / GC may
+            // be able to figure out that we do not need the trackedObject anymore and so already enqueue it for
+            // collection before we actually get a chance to close the enclosing ResourceLeak.
+            boolean closed = close() && trackedObject != null;
+
+            // Ensure that the object that was tracked is the same as the one that was passed to close(...).
+            assert trackedHash == System.identityHashCode(trackedObject);
```

Comment:
@normanmaurer , I see. Would you probably add an explanation message then to the `assert` statement?